data_diff/dbt.py

```diff
@@ -302,15 +323,72 @@ def _cloud_diff(diff_vars: DiffVars) -> None:
             send_event_json(event_json)
 
         if error:
-            raise error
+            logger.error(error)
+
+
+def _setup_cloud_diff() -> Tuple[Optional[str]]:
+    datafold_host = os.environ.get("DATAFOLD_HOST")
+    if datafold_host is None:
+        datafold_host = "https://app.datafold.com"
+    datafold_host = datafold_host.rstrip("/")
+    rich.print(f"Cloud datafold host: {datafold_host}\n")
+    url = f"{datafold_host}/api/v1/datadiffs"
+
+    api_key = os.environ.get("DATAFOLD_API_KEY")
+    if not api_key:
+        rich.print("[red]API key not found, add it as an environment variable called DATAFOLD_API_KEY.")
+        yes_or_no = Confirm.ask("Would you like to generate a new API key?")
+        if yes_or_no:
+            webbrowser.open(f"{datafold_host}/login?next={datafold_host}/users/me")
+            return None, None, None
+        else:
+            raise ValueError("Cannot diff because the API key is not provided")
+
+    return datafold_host, url, api_key
+
+
+def _cloud_submit_diff(url, payload, headers) -> str:
+    response = requests.request("POST", url, headers=headers, json=payload, timeout=30)
+    response.raise_for_status()
+    response_json = response.json()
+    diff_id = str(response_json["id"])
+
+    if diff_id is None:
+        raise Exception(f"Api response did not contain a diff_id: {str(response_json)}")
+    return diff_id
+
+
+def _cloud_poll_and_get_summary_results(url, headers):
+    summary_results = None
+    start_time = time.time()
+    sleep_interval = 5  # starts at 5 sec
+    max_sleep_interval = 60
+    max_wait_time = 300
+
+    while not summary_results:
+        response = requests.request("GET", url, headers=headers, timeout=30)
+        response.raise_for_status()
+        response_json = response.json()
+
+        if response_json["status"] == "success":
+            summary_results = response_json
+        elif response_json["status"] == "failed":
+            raise Exception(f"Diff failed: {str(response_json)}")
+
+        if time.time() - start_time > max_wait_time:
+            raise Exception("Timed out waiting for diff results")
+
+        time.sleep(sleep_interval)
+        sleep_interval = min(sleep_interval * 2, max_sleep_interval)
+
+    return summary_results
```

Comment:
wasn't sure but these seemed like pretty safe values, might lean too much towards timeouts

```
    sleep_interval = 5  # starts at 5 sec
    max_sleep_interval = 60
    max_wait_time = 300
```

Reply:
Since the url is printed with the error, I think we're okay here for now. Can always adjust:
![image](https://user-images.githubusercontent.com/11282254/229225104-a9c371fb-b2ab-4685-9030-3d7223037914.png)